app/models/binary/explore/N5MultiscalesExplorer.scala

```diff
@@ -0,0 +1,129 @@
+    for {
+      zattrsPath <- Fox.successful(remotePath.resolve(N5Metadata.FILENAME_ATTRIBUTES_JSON))
+      n5Metadata <- parseJsonFromPath[N5Metadata](zattrsPath) ?~> s"Failed to read OME NGFF header at $zattrsPath"
+      layers <- Fox.serialCombined(n5Metadata.multiscales)(layerFromN5MultiscalesItem(_, remotePath, credentials))
+    } yield layers
+
+  private def layerFromN5MultiscalesItem(multiscalesItem: N5MultiscalesItem,
+                                         remotePath: Path,
+                                         credentials: Option[FileSystemCredentials]): Fox[(N5Layer, Vec3Double)] =
+    for {
+      voxelSizeNanometers <- extractVoxelSize(multiscalesItem.datasets.map(_.transform))
+      magsWithAttributes <- Fox.serialCombined(multiscalesItem.datasets)(d =>
+        n5MagFromDataset(d, remotePath, voxelSizeNanometers, credentials))
+      _ <- bool2Fox(magsWithAttributes.nonEmpty) ?~> "zero mags in layer"
+      elementClass <- elementClassFromMags(magsWithAttributes) ?~> "Could not extract element class from mags"
+      boundingBox = boundingBoxFromMags(magsWithAttributes)
+      name <- guessNameFromPath(remotePath)
+      layer: N5Layer = if (looksLikeSegmentationLayer(name, elementClass)) {
+        N5SegmentationLayer(name, boundingBox, elementClass, magsWithAttributes.map(_.mag), largestSegmentId = None)
+      } else N5DataLayer(name, Category.color, boundingBox, elementClass, magsWithAttributes.map(_.mag))
+    } yield (layer, voxelSizeNanometers)
+
+  private def extractAxisOrder(axes: List[String]): Fox[AxisOrder] = {
+    val x = axes.indexWhere(_ == "x")
+    val y = axes.indexWhere(_ == "y")
+    val z = axes.indexWhere(_ == "z")
+    val c = axes.indexWhere(_ == "c")
+
+    val cOpt = if (c == -1) None else Some(c)
+    for {
+      _ <- bool2Fox(x >= 0 && y >= 0 && z >= 0) ?~> s"invalid xyz axis order: $x,$y,$z."
+    } yield AxisOrder(x, y, z, cOpt)
+  }
+
+  private def extractAxisUnitFactors(units: List[String], axisOrder: AxisOrder): Fox[Vec3Double] =
+    for {
+      xUnitFactor <- spaceUnitToNmFactor(units(axisOrder.x))
+      yUnitFactor <- spaceUnitToNmFactor(units(axisOrder.y))
+      zUnitFactor <- spaceUnitToNmFactor(units(axisOrder.z))
+    } yield Vec3Double(xUnitFactor, yUnitFactor, zUnitFactor)
+
+  private def spaceUnitToNmFactor(unit: String): Fox[Double] =
+    unit.toLowerCase match {
+      case ""          => Fox.successful(1.0)
+      case "ym"        => Fox.successful(1e-15)
+      case "zm"        => Fox.successful(1e-12)
+      case "am"        => Fox.successful(1e-9)
+      case "fm"        => Fox.successful(1e-6)
+      case "pm"        => Fox.successful(1e-3)
+      case "nm"        => Fox.successful(1.0)
+      case "µm"        => Fox.successful(1e3)
+      case "mm"        => Fox.successful(1e6)
+      case "cm"        => Fox.successful(1e7)
+      case "dm"        => Fox.successful(1e8)
+      case "m"         => Fox.successful(1e9)
```

Comment:
Maybe this can be shared with the other unit definitions? Or are those too different?

Reply:
From what I’ve seen, ngff uses `nanometer` while n5 uses `nm` so they are pretty distinct. Of course, we could still merge the matching into one big function, what would you prefer?

Reply:
Let's keep it separate as-is for now :+1:
The only reference I found is https://github.com/saalfeldlab/n5-viewer/blob/23f86b32e77ff1856893fa955523fb016cef3d82/README.md#container-structure,
where it says that not setting anything assumes `µm`, I think that is currently not the case. Also, it specifically shows the string `um`, which is missing in this list.